include/crm/common/results.h

```diff
@@ -0,0 +1,127 @@
+ * caught on. It uses 0 and 64-78.
+ *
+ * Bash reserves 2 ("incorrect builtin usage") and 126-255 (126 is "command
+ * found but not executable", 127 is "command not found", 128 + n is
+ * "interrupted by signal n").
+ *
+ * tldp.org recommends 64-113 for application use.
+ *
+ * We try to overlap with the above conventions when practical.
+ */
+typedef enum crm_exit_e {
+    // Common convention
+    CRM_EX_OK                   =   0,
+    CRM_EX_ERROR                =   1,
+
+    // LSB + OCF
+    CRM_EX_INVALID_PARAM        =   2,
+    CRM_EX_UNIMPLEMENT_FEATURE  =   3,
+    CRM_EX_INSUFFICIENT_PRIV    =   4,
+    CRM_EX_NOT_INSTALLED        =   5,
+    CRM_EX_NOT_CONFIGURED       =   6,
+    CRM_EX_NOT_RUNNING          =   7,
+
+    // sysexits.h
+    CRM_EX_USAGE                =  64, // command line usage error
+    CRM_EX_DATAERR              =  65, // user-supplied data incorrect
+    CRM_EX_NOINPUT              =  66, // input file not available
+    CRM_EX_NOUSER               =  67, // user does not exist
+    CRM_EX_NOHOST               =  68, // host unknown
+    CRM_EX_UNAVAILABLE          =  69, // needed service unavailable
+    CRM_EX_SOFTWARE             =  70, // internal software bug
+    CRM_EX_OSERR                =  71, // external (OS/environmental) problem
+    CRM_EX_OSFILE               =  72, // system file not usable
+    CRM_EX_CANTCREAT            =  73, // file couldn't be created
+    CRM_EX_IOERR                =  74, // file I/O error
+    CRM_EX_TEMPFAIL             =  75, // try again
+    CRM_EX_PROTOCOL             =  76, // protocol violated
+    CRM_EX_NOPERM               =  77, // non-file permission issue
+    CRM_EX_CONFIG               =  78, // misconfiguration
+
+    // Custom
+    CRM_EX_FATAL                = 100, // do not respawn
+    CRM_EX_PANIC                = 101, // panic the local host
+    CRM_EX_DISCONNECT           = 102, // lost connection to something
+    CRM_EX_OLD                  = 103, // update older than existing config
+    CRM_EX_DIGEST               = 104, // digest comparison failed
+    CRM_EX_NOSUCH               = 105, // requested item does not exist
+    CRM_EX_QUORUM               = 106, // local partition does not have quorum
+    CRM_EX_UNSAFE               = 107, // requires --force or new conditions
+    CRM_EX_EXISTS               = 108, // requested item already exists
+    CRM_EX_MULTIPLE             = 109, // requested item has multiple matches
+
+    // Other
+    CRM_EX_TIMEOUT              = 124, // convention from timeout(1)
+    CRM_EX_MAX                  = 255, // ensure crm_exit_t can hold this
+} crm_exit_t;
+
+const char *pcmk_strerror(int rc);
+const char *pcmk_errorname(int rc);
+const char *bz2_strerror(int rc);
+crm_exit_t crm_errno2exit(int rc);
+const char *crm_exit_name(crm_exit_t exit_code);
+const char *crm_exit_str(crm_exit_t exit_code);
+crm_exit_t crm_exit(crm_exit_t rc);
+
+#endif
```

Comment:
Neat! Like it. Thanks. I probably wouldn't mind to still use the filename error.h given that actually only crm_exit() is not directly about errors. But of course it makes sense to be always precise ;)

Reply:
Yep, I was going to keep it as error.h, but having pcmk_ok in error.h was bugging me ;)